hl7_receiving.py

```
@@ -67,39 +65,70 @@ def get_file_name(data: str):
 
     if data.startswith(MLLP_START) and data.endswith(MLLP_END):
         data = data[1:-2]
-    message = data.decode().replace('\n', '\r')
+    message = data.decode("latin-1").replace('\n', '\r')
 
-    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S%f")
 
     try:
         m = parse_message(message, find_groups=False)
     except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)